### PR TITLE
fix: add back CJS support, with a test

### LINKED ISSUE #151: 🐛 Bug: CJS build of v2.1.1 imports an ESM-only module
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/omnidan/node-emoji/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

Importing the CJS build should work correctly.

### Actual

A node error is thrown:

```
Error [ERR_REQUIRE_ESM]: require() of ES Module /workdir/node_modules/node-emoji/node_modules/@sindresorhus/is/dist/index.js from /workdir/node_modules/node-emoji/lib/index.cjs not supported.
```

### Additional Info

[`@sindresorhus/is` went ESM-only in v5](https://github.com/sindresorhus/is/releases/tag/v5.0.0), making it incompatible in a CJS build without some kind CJS-transpilation in place (don't think this is the case with this library).

The upgrade to `@sindresorhus/is@6` was done [here](https://github.com/omnidan/node-emoji/pull/142/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R61).

### LINKED ISSUE #153: 🛠 Tooling: Add test that CJS require() works (not just ESM import)
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/omnidan/node-emoji/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

This library is meant to support both CommonJS (CJS / `require()`) and ECMAScript Modules (ESM / `import`) entry points:

https://github.com/omnidan/node-emoji/blob/6714e2102acf0cf6a556f168952f90582b1a1663/package.json#L30-L35

But, per #151, it's possible for a change to make it in that breaks just CJS. We should add in some kind of end-to-end/integration test to make sure that CJS isn't broken.

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #151; fixes #153
- [x] That issue was marked as [`status: accepting prs`](https://github.com/omnidan/node-emoji/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/omnidan/node-emoji/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Downgrades `@sindresorhus/is` down to v4, its last major version that still supported CommonJS.

Adds a test with a `.cjs` extension and `require` that will fail if CommonJS support is broken again.

Also adds explicit `exports.types` to `package.json`, as I've had to do that in past packages.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,14 +56,15 @@
...
     "tsc": "tsc"
   },
   "lint-staged": {
     "*": "prettier --ignore-unknown --write"
   },
   "dependencies": {
-    "@sindresorhus/is": "^6.0.0",
-    "char-regex": "^2.0.0",
+    "@sindresorhus/is": "^4.6.0",
+    "char-regex": "^1.0.2",
```
Comment (JoshuaKGoldberg) [inline package.json:67]: #149 also broke CJS support with `char-regex@2`.

**Comment by JoshuaKGoldberg** [comment]
FYI @omnidan - I can merge this in tomorrow if you don't have a chance to look (to unbreak what I broke over the weekend)

File: src/emojify.ts [hunk 2]
```diff
@@ -21,9 +21,9 @@ export const emojify = (
   const fallbackFunction =
     fallback === undefined ? fallback : asFunction(fallback)
 
-  assert.string(input)
-  assert.any([is.undefined, is.function], fallbackFunction)
-  assert.function(format)
+  is.assert.string(input)
+  is.assert.any([is.default.undefined, is.default.function_], fallbackFunction)
```
Comment (JoshuaKGoldberg) [inline src/emojify.ts:25]: I played around with these a bit but couldn't find anything cleaner. Ran out of time today. But not super please about using a `.default.` 🤔

**Comment by omnidan** [comment]
@JoshuaKGoldberg please do so, I quickly checked and, while it is annoying that we have to downgrade, it is fine for now. We will have to look at dropping CJS support in the future I guess, but for now I believe it's still too frequently used.

**Comment by omnidan** [comment]
I think we have to stop the renovate bot from upgrading those packages though 😅

**Comment by JoshuaKGoldberg** [comment]
Yeah a test that fails on CJS breakages should stop it. Renovate can also be told to hush for particular packages

**Pull request merged**